galleon-plugins/src/main/java/org/wildfly/galleon/plugin/ShadedModel.java

```diff
@@ -0,0 +1,260 @@
+        this.channelArtifactResolution = channelArtifactResolution;
+        this.recorder = recorder;
+    }
+
+    public List<MavenArtifact> getArtifacts() throws ProvisioningException, IOException {
+        List<MavenArtifact> artifacts = new ArrayList<>();
+        Element shadedDependencies = rootElement.getFirstChildElement("shaded-dependencies",
+                rootElement.getNamespaceURI());
+        Elements dependencies = shadedDependencies.getChildElements();
+        for (int i = 0; i < dependencies.size(); i++) {
+            Element e = dependencies.get(i);
+            String[] coords = e.getValue().split(":");
+            String ga = coords[0] + ":" + coords[1];
+            MavenArtifact a = Utils.toArtifactCoords(mergedArtifactVersions, e.getValue(), false, channelArtifactResolution, requireChannel);
+            artifactResolver.resolve(a);
+            if (log.isVerboseEnabled()) {
+                log.verbose("Shadel model dependency: " + e.getValue() + " resolved version " + a.getVersion());
+            }
+            Path transformed = installer.installCopiedArtifact(a);
+            a.setPath(transformed);
+            artifacts.add(a);
+            if (recorder.isPresent()) {
+                recorder.get().cache(a, a.getPath());
+            }
+        }
+        return artifacts;
+    }
+
+    public Map<String, String> getManifestEntries() {
+        Map<String, String> entries = new HashMap<>();
+        Element manifestEntries = rootElement.getFirstChildElement("manifestEntries",
+                rootElement.getNamespaceURI());
+        Elements elements = manifestEntries.getChildElements();
+        for (int i = 0; i < elements.size(); i++) {
+            Element e = elements.get(i);
+            entries.put(e.getLocalName(), e.getValue());
+        }
+        return entries;
+    }
+
+    public String getMainClass() {
+        String ret = null;
+        Element mainClass = rootElement.getFirstChildElement("main-class",
+                rootElement.getNamespaceURI());
+        if (mainClass != null) {
+            ret = mainClass.getValue();
+        }
+        return ret;
+    }
+
+    public String getName() {
+        return rootElement.getFirstChildElement("name",
+                rootElement.getNamespaceURI()).getValue();
+    }
+
+    public void buildJar(Path shadedJar) throws IOException, ProvisioningException {
+        if (log.isVerboseEnabled()) {
+            log.verbose("Assembling shaded jar " + shadedJar);
+        }
+        Path tmpTarget = tmpPath.resolve("assemble_target").resolve(shadedJar.getFileName());
+        Files.createDirectories(tmpTarget);
+        for (MavenArtifact dependency : getArtifacts()) {
+            Utils.navigateArtifact(dependency.getPath(), tmpTarget, this);
+        }
+        update(tmpTarget);
+        ZipUtils.zip(tmpTarget, shadedJar);
+        IoUtils.recursiveDelete(tmpTarget);
+    }
+
+    public void update(Path target) throws IOException {
```

Comment:
nitpick - how about naming this method something like `generateMetaInf` or `updateMetaInf`? Think it might help understanding what's happening at a glance.

Also does it need to be public, I think it's only used locally?